olive/passes/pytorch/lora.py

```diff
@@ -214,20 +215,62 @@ def validate_search_point(
     ) -> bool:
         if with_fixed_value:
             search_point = self.config_at_search_point(search_point or {})
-        if search_point.get("use_ort_trainer"):
-            if search_point.get("torch_dtype") == "bfloat16":
-                logger.info(
-                    "bfloat16 is not supported by onnxruntime-training yet. Please use a different torch_dtype."
-                )
-                return False
-            if search_point.get("training_args", {}).get("gradient_checkpointing"):
-                logger.info(
-                    "gradient_checkpointing is not supported by onnxruntime-training. Please set gradient_checkpointing"
-                    " to False."
-                )
-                return False
+        if search_point.get("use_ort_trainer") and search_point.get("training_args", {}).get("gradient_checkpointing"):
+            logger.info(
+                "gradient_checkpointing is not supported by onnxruntime-training. Please set gradient_checkpointing"
+                " to False."
+            )
+            return False
         return True
 
+    @classmethod
+    def check_dependencies(cls, config: ConfigBase, is_qlora: bool = False):
+        """Check dependencies for the pass."""
+        if config.use_ort_trainer:
+            # check for ort trainer dependencies
+            try:
+                from optimum.onnxruntime import ORTTrainer  # noqa: F401
+                from optimum.onnxruntime.utils import is_onnxruntime_training_available
+                from torch_ort import ORTModule  # noqa: F401
+
+                assert is_onnxruntime_training_available(), "onnxruntime-training is not available."
+            except (ImportError, AssertionError):
+                raise RuntimeError(
```

Comment:
>RuntimeError [](http://example.com/codeflow?start=22&length=12)

RuntimeError seems not in EXCEPTIONS_TO_RAISE, do we need add it into the list?

Reply:
That is a good question. I changed the error here to `ImportError` since that what rest of Olive uses for such version and dependency related errors. For `RuntimeError`, I think we might have to review separately later if needed.